SCons/Tool/MSCommon/MSVC/ScriptArguments.py

```diff
@@ -574,57 +613,68 @@ def _msvc_version_toolset_vcvars(msvc, vc_dir, toolset_version):
 
 def _msvc_script_argument_toolset_constraints(msvc, toolset_version):
 
-    if msvc.vs_def.vc_buildtools_def.vc_version_numeric < VS2017.vc_buildtools_def.vc_version_numeric:
+    if msvc.vs_def.vc_buildtools_def.msvc_version_numeric < VS2017.vc_buildtools_def.msvc_version_numeric:
```

Comment:
You have several instances of this condition, might be clearer to just have a simple function to do this?

Reply:
At present, I'm not sure it would clearer to move to a function.  For some, it may be harder to comprehend what exactly is being compared.

While long-ish dereference chains might not be everyone's cup of tea, as written, they are simple one-line numeric comparisons.  The source of the resolved values are immediately apparent by inspection.

What is getting increasingly complicated to comprehend is *which* of the many msvc *version numbers* are being used.

In local prototypes, judiciously adding properties to some of the data structures could be an effective method of visually reducing the dereference chains.  The idea is that each property might resolve one or more dereferences.  Basically, making a derived/embedded field appear as if it is a local attribute.

Something like:
```
msvc.vs_def.vc_buildtools_def.msvc_version_numeric < 2017.vc_buildtools_def.msvc_version_numeric
```
becomes:
```
msvc.vs_def.msvc_version_numeric < 2017.msvc_version_numeric
```

Unfortunately, this causes a non-trivial increase in the number of lines of code.

*If directed, moving some the comparisons to functions could be straightforward.  I'm not wild about it though.*